src/Momento.Sdk/Internal/ScsTopicClient.cs

```diff
@@ -0,0 +1,183 @@
+    }
+
+    public async Task<TopicPublishResponse> Publish(string cacheName, string topicName, byte[] value)
+    {
+        var topicValue = new _TopicValue
+        {
+            Binary = value.ToByteString()
+        };
+        return await SendPublish(cacheName, topicName, topicValue);
+    }
+
+    public async Task<TopicPublishResponse> Publish(string cacheName, string topicName, string value)
+    {
+        var topicValue = new _TopicValue
+        {
+            Text = value
+        };
+        return await SendPublish(cacheName, topicName, topicValue);
+    }
+
+    public async Task<TopicSubscribeResponse> Subscribe(string cacheName, string topicName,
+        ulong? resumeAtTopicSequenceNumber = null)
+    {
+        return await SendSubscribe(cacheName, topicName, resumeAtTopicSequenceNumber);
+    }
+
+    private const string RequestTypeTopicPublish = "TOPIC_PUBLISH";
+
+    private async Task<TopicPublishResponse> SendPublish(string cacheName, string topicName, _TopicValue value)
+    {
+        _PublishRequest request = new _PublishRequest
+        {
+            CacheName = cacheName,
+            Topic = topicName,
+            Value = value
+        };
+
+        try
+        {
+            _logger.LogTraceExecutingTopicRequest(RequestTypeTopicPublish, cacheName, topicName);
+            await grpcManager.Client.publish(request, new CallOptions(deadline: CalculateDeadline()));
+        }
+        catch (Exception e)
+        {
+            return _logger.LogTraceTopicRequestError(RequestTypeTopicPublish, cacheName, topicName,
+                new TopicPublishResponse.Error(_exceptionMapper.Convert(e)));
+        }
+
+        return _logger.LogTraceTopicRequestSuccess(RequestTypeTopicPublish, cacheName, topicName,
+            new TopicPublishResponse.Success());
+    }
+
+    private async Task<TopicSubscribeResponse> SendSubscribe(string cacheName, string topicName,
+        ulong? resumeAtTopicSequenceNumber)
+    {
+        var request = new _SubscriptionRequest
+        {
+            CacheName = cacheName,
+            Topic = topicName
+        };
+        if (resumeAtTopicSequenceNumber != null)
+        {
+            request.ResumeAtTopicSequenceNumber = resumeAtTopicSequenceNumber.Value;
+        }
+
+        AsyncServerStreamingCall<_SubscriptionItem> subscription;
+        try
+        {
+            _logger.LogTraceExecutingTopicRequest(RequestTypeTopicPublish, cacheName, topicName);
+            subscription = grpcManager.Client.subscribe(request, new CallOptions());
+        }
+        catch (Exception e)
+        {
+            return _logger.LogTraceTopicRequestError(RequestTypeTopicPublish, cacheName, topicName,
+                new TopicSubscribeResponse.Error(_exceptionMapper.Convert(e)));
+        }
+
+        var response = new TopicSubscribeResponse.Subscription(
+            token => MoveNextAsync(subscription, token, cacheName, topicName),
```

Comment:
I'm passing in closures here so I can encapsulate the grpc message handling code.